### PR TITLE
AVI-18: Implement the login page

### LINKED ISSUE #18: Implement the login page
### Description

As a user, I should be able to login to the application.

### Acceptance Criteria

- [x] Implement login page using React
- [x] Create re-usable components for things like (input, buttons etc)
- [x] Add inputs/form validation
- [ ] Upon login, user should be redirected to a blank dashboard page

### NOTE

For now, don't worry about making a HTTP request upon form submission. We'll handle the API integration later 👌

### PR DESCRIPTION
**Pull request by Doyin02**
### What does this PR do

This is the login page implemented with React

Closes #18

### How should it be tested

- Go to your local development
- Note: Inside the the component folder, you will find re-usable components (Button and Input)
- Note: Also inside the feature folder, you will find the login.jsx file.
- Open the terminal, and launch the REACT app with npm or yarn start.

### CODE REVIEW

**Comment by harry-olabs** [comment]
@Doyin02 work on the following:

- Remove the scale effect on the login form card when hovered
- Remove the scale(0.9), it is not needed
- Update the login form input padding to 14px
- I see you styled the `body` in the login scss file, it might overlap and effect the styling for other pages. You avoid doing that. Instead, make use of a container/wrapper, then style the container.
- When I click on the login button, I don't get redirected to the Dashboard page. 
- It looks like you are not loading Google fonts, the UI is using sans-serif as opposed to Monsterrat.

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+  return <Routes>
+    <Route path="/" element={<Login/>}/>
+    <Route path="/registration" element={<Registration/>}/>
+  </Routes>
```
Comment (harry-olabs) [inline src/App.js]: Move this to a dedicated route file

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
 }
 
 export default App;
+(
+  <div className="App">
+    {/* <Login /> */}
+    <Registration/>
+  </div>
+);
```
Comment (harry-olabs) [inline src/App.js]: This must be typo, remove it

File: src/components/Button/Button.css [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    background-color: #4caf50;
+    color: #fff;
+    cursor: pointer;
+    border: none;
+    padding: 12px 20px;
+    font-size: 16px;
+    border-radius: 5px;
+    transition: background-color 0.3s ease;
+    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
+}
```
Comment (harry-olabs) [inline src/components/Button/Button.css]: Apply proper formatting to this stylesheet

File: src/components/Input/Input.css [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  border-radius: 5px;
+  border: 1px solid #ccc;
+  background-color: #f2f2f2;
+  transition: background-color 0.3s ease;
+  font-size: 16px;
+
+}
+input:hover {
+  background-color: #e0e0e0;
+}
```
Comment (harry-olabs) [inline src/components/Input/Input.css]: Let there be space (empty line) after each style block 🔥

File: src/components/Button/index.jsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import "./Button.css"
+
+const Button = ({text}) => {
+  return (
+    <button className='button'>{text}</button>
```
Comment (harry-olabs) [inline src/components/Button/index.jsx]: Make use of a custom-class name (e.g `custom-btn`) as opposed to generic name `button`.

Comment (harry-olabs) [reply src/components/Button/index.jsx]: A button should ideally have an `onClick` handler prop. Be sure to add it

File: src/components/Input/index.jsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        onChange(inputValue);
+      }
+    } else if (type === "repeat-password") {
+      if (inputValue !== value) {
+        setError("Passwords do not match.");
+      } else {
+        setError("");
+        onChange(inputValue);
+      }
+    }
```
Comment (harry-olabs) [inline src/components/Input/index.jsx]: Move this to a function, then execute the function within the handleChange.

File: src/feature/Login/Login.module.css [hunk 7]
```diff
@@ -0,0 +1,99 @@
+body {
+  font-family: "Montserrat", sans-serif;
```
Comment (harry-olabs) [inline src/feature/Login/Login.module.css]: We have variable for this, make use of it.

File: src/feature/Login/index.jsx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+            placeholder="Enter your password"
+          />
+        </div>
+        <Button text="Login" />
+      </form>
+      <div className={styles.register__link}>
+        <p>
+          Don't have an account yet?
+          <span>
+          <Link to="/Registration" className={styles.register__link__anchor}>Register</Link>
```
Comment (harry-olabs) [inline src/feature/Login/index.jsx]: Route path should be in lowercase. And for multiple words, make use hyphen to separate the words (e.g `my-profile`).

File: src/index.js [hunk 9]
```diff
@@ -1,13 +1,16 @@
...
 import App from './App';
 import reportWebVitals from './reportWebVitals';
 
+
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <React.StrictMode>
-    <App />
+    
+   <BrowserRouter><App/></BrowserRouter>
```
Comment (harry-olabs) [inline src/index.js]: Format this properly to improve readability

File: template/login.html
Comment (harry-olabs) [inline template/login.html:1]: You shouldn't have to touch this file. All you need to do was to copy the codes you need.

— **IMPORTANT**: Never remove anything from the template files unless you want update the UI template.

**Review by harry-olabs — CHANGES_REQUESTED** [review]
@Doyin02 It is almost looking good. Attend to the following requested changes

File: src/feature/Login/index.jsx (hunk 8 above)
Comment (harry-olabs) [inline src/feature/Login/index.jsx]: The class name `register__link__anchor` should ideally be `login_link...`

File: src/components/Button/Button.css [hunk 10]
```diff
@@ -0,0 +1,17 @@
+* {
+  padding: 0;
+  margin: 0;
+  box-sizing: border-box;
+}
```
Comment (harry-olabs) [inline src/components/Button/Button.css]: This should not be here, but rather in the root level like `index.css` file — Remove it.

File: src/components/Button/index.jsx [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import "./Button.css";
+
+const Button = ({ to, text, handleClick }) => {
```
Comment (harry-olabs) [inline src/components/Button/index.jsx]: The `to` prop is not used. Remove it

File: src/components/Input/Input.css [hunk 12]
```diff
@@ -0,0 +1,20 @@
+* {
+  padding: 0;
+  margin: 0;
+  box-sizing: border-box;
+}
```
Comment (harry-olabs) [inline src/components/Input/Input.css]: I've commented about this up top, remove it 🔥

File: src/components/Main-styles/variables.css
Comment (harry-olabs) [inline src/components/Main-styles/variables.css:1]: - Move this to the root, preferably in a `styles` folder.
- Also, I don't see the `variables.css` file in the template styles folder anymore. Did you remove it? If so, put it back.

File: src/components/Main-styles/main.css
Comment (harry-olabs) [inline src/components/Main-styles/main.css:1]: Move this to the root, preferably in a `styles` folder.

File: src/App.js [hunk 13]
```diff
@@ -1,25 +1,11 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    
+      <AppRoutes />
+
```
Comment (harry-olabs) [inline src/App.js]: This is not formatted properly, correct it

**Review by harry-olabs — CHANGES_REQUESTED** [review]
@Doyin02 we are almost there. Attend to the comments.

**IMPORTANT**: Don't forget to mark a comment as resolved whenever you resolve it.

File: src/feature/Login/index.jsx (hunk 8 above)
Comment (harry-olabs) [reply src/feature/Login/index.jsx]: Reach out if you don't understand what you're expected to do here

File: src/components/Main-styles/variables.css
Comment (Doyin02) [reply src/components/Main-styles/variables.css:1]: when I tried to access the varable.css file from the react app, it gave an error that it cannot access any file that is outside of the src folder, that was why I brought it inside the src folder

Comment (harry-olabs) [reply src/components/Main-styles/variables.css:1]: You are not expected to make use of the `variables.css` file in the template folder.

Comment (Doyin02) [reply src/components/Main-styles/variables.css:1]: How am I going to access the styles in there

Comment (harry-olabs) [reply src/components/Main-styles/variables.css:1]: Like I mentioned in a previous comment, don't move or delete anything in the `template` folder. You can only reference the codes/styles

Comment (Doyin02) [reply src/components/Main-styles/variables.css:1]: I don't understand sir, the components brought out an error when I wanted to access the variable.css, it said I cannot access anything outside the src folder, what am I supposed to do?

Comment (harry-olabs) [reply src/components/Main-styles/variables.css:1]: @Doyin02 
- Firstly, keep this `variables.css` file, but be sure to move it to the root in `styles` folder.
- Secondly, put back the `variables.css` file into the `template/styles` folder.

Comment (Doyin02) [reply src/components/Main-styles/variables.css:1]: Do yo mean I can leave the variable.css file that is inside the Main styles folder, but I should just move the variable.css to the styles also?

File: src/components/Button/Button.css [hunk 14]
```diff
@@ -0,0 +1,11 @@
+.custom-btn {
+  background-color: #4caf50;
```
Comment (harry-olabs) [inline src/components/Button/Button.css]: Replace color value with variable

File: src/components/Button/Button.css [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+  color: #fff;
+  cursor: pointer;
+  border: none;
+  padding: 12px 20px;
+  font-size: 16px;
+  border-radius: 5px;
+  transition: background-color 0.3s ease;
+  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
```
Comment (harry-olabs) [inline src/components/Button/Button.css]: I believe there is a variable for box-shadow, make use of it

File: src/components/Input/Input.css [hunk 16]
```diff
@@ -0,0 +1,14 @@
+ input {
+  width: 100%;
+  padding: 14px;
+  color: rgb(72, 68, 68);
+  border-radius: 5px;
+  border: 1px solid #ccc;
+  background-color: #f2f2f2;
```
Comment (harry-olabs) [inline src/components/Input/Input.css:7]: Make use of variable

File: src/components/Input/Input.css [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+  transition: background-color 0.3s ease;
+  font-size: 16px;
+}
+
+input:hover {
+  background-color: #e0e0e0;
```
Comment (harry-olabs) [inline src/components/Input/Input.css]: Make use of variable

File: src/components/Input/Input.css [hunk 18]
```diff
@@ -0,0 +1,14 @@
+ input {
+  width: 100%;
+  padding: 14px;
+  color: rgb(72, 68, 68);
```
Comment (harry-olabs) [inline src/components/Input/Input.css]: Make use of variable

File: src/feature/Login/Login.module.css [hunk 19]
```diff
@@ -0,0 +1,100 @@
+@import "/src/styles/main.css";
+
+.body {
+  background: linear-gradient(135deg, #00f260, #0575e6);
```
Comment (harry-olabs) [inline src/feature/Login/Login.module.css]: Make use of variable

File: src/feature/Login/Login.module.css [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+  margin-bottom: 5px;
+}
+
+.form__input[type="email"],
+.form__input[type="password"] {
+  width: 100%;
+  padding: 12px;
+  border-radius: 5px;
+  border: 1px solid #ccc;
+  background-color: #f2f2f2;
```
Comment (harry-olabs) [inline src/feature/Login/Login.module.css]: Make use of variable

File: src/feature/Login/Login.module.css [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+  transition: background-color 0.3s ease;
+}
+
+.form__input[type="email"]:hover,
+.form__input[type="password"]:hover {
+  background-color: #e0e0e0;
```
Comment (harry-olabs) [inline src/feature/Login/Login.module.css]: Make use of variable

File: src/feature/Login/Login.module.css [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+.form__group {
+  margin-bottom: 20px;
+}
+
+.form__label {
+  display: block;
+  font-weight: var(--font-bold);
+  color: #333;
```
Comment (harry-olabs) [inline src/feature/Login/Login.module.css]: Make use of variable

File: src/feature/Login/Login.module.css [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+.login__link {
+  text-align: center;
+  margin-top: 20px;
+  color: var(--primary-black);
+}
+
+.login__link__anchor {
+  color: #333;
```
Comment (harry-olabs) [inline src/feature/Login/Login.module.css]: Make use of variable

File: src/index.css [hunk 24]
```diff
@@ -1,13 +1,19 @@
+@import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@400;500;600;700&display=swap");
+
 body {
   margin: 0;
-  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
-    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
-    sans-serif;
+  font-family: "Montserrat", sans-serif;
```
Comment (harry-olabs) [inline src/index.css]: You are supposed to make use of `--font-family` variable. Correct this

File: src/index.css [hunk 25]
```diff
@@ -1,13 +1,19 @@
...
+  margin: 0;
+  box-sizing: border-box;
+}
+
 
 code {
-  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
+  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
     monospace;
 }
```
Comment (harry-olabs) [inline src/index.css]: You can remove this style block, it is not used

File: src/styles/variables.css [hunk 26]
```diff
@@ -3,6 +3,8 @@
   --primary-red: #d15267;
   --bg-color: white;
   --primary-white: white;
+  --primary-green: #45a049;
+  --primary-black:black;
```
Comment (harry-olabs) [inline src/styles/variables.css]: Format properly (space after colon)

**Review by harry-olabs — CHANGES_REQUESTED** [review]
@Doyin02 It's almost looking good, attend to the comments

**Review by harry-olabs — APPROVED** [review]
LGTM 🚀

**Pull request merged**